File: problem_283.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
    index = 0
    zero_count = 0
    while !nums[index].nil? do
        if nums[index] == 0
            zero_count += 1
            nums.delete_at(index)
        else
            index += 1
        end
    end

    zero_count.times do
        nums << 0
    end
    nums
end

nums = [0,1,0,3,12]
p move_zeroes(nums)

File: problem_283.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut index = 0;
        let mut zero_count = 0;
        let len = nums.len();
​
        while index < len - zero_count {
            if nums[index] == 0 {
                zero_count += 1;
                nums.remove(index);
            } else {
                index += 1;
            }
        }
​
        for _ in 0..zero_count { nums.push(0) }
    }
}
